src/Datagrid/Pager.php

```diff
@@ -80,16 +83,18 @@ public function init()
         }
     }
 
-    private function getColumnsForQuery($rootAlias)
+    /**
+     * @return ProxyQueryInterface
+     */
+    private function getClonedQuery()
     {
-        if (1 === \count($countColumns = $this->getCountColumn())) {
-            return $rootAlias.'.'.current($countColumns);
-        }
+        /** @var ProxyQueryInterface $cloneQuery */
+        $cloneQuery = clone $this->getQuery();
 
-        $columns = implode(",'-',", array_map(function ($column) use ($rootAlias) {
-            return "$rootAlias.$column";
-        }, $countColumns));
+        if (\count($this->getParameters()) > 0) {
```

Comment:
Since `$this->getParameters()` returns an array, I think there is no need for `count()`.

Reply:
I think this is a check if the array is not empty

Reply:
In a [loose check](https://www.php.net/manual/en/types.comparisons.php), an empty array already is evaluated as `false`.

Reply:
implicit conversion is not good idea

Reply:
AFAIK, there is no conversion, just comparison.

Reply:
I agree with both of you, but this sounds minor to me and not worth the discussion, as both ways are ok with the same result

Reply:
My concern is just about saving a non required function call. Is just for performance reasons. But since the call was already there, it's fine for me if we decide to keep it.